service/history/task/priority_assigner.go

```diff
@@ -39,20 +39,16 @@ var (
 	lowTaskPriority     = common.GetTaskPriority(common.LowPriorityClass, common.DefaultPrioritySubclass)
 )
 
-type (
-	priorityAssignerImpl struct {
-		sync.RWMutex
-
-		currentClusterName string
-		domainCache        cache.DomainCache
-		config             *config.Config
-		logger             log.Logger
-		scope              metrics.Scope
-		rateLimiters       *quotas.Collection
-	}
-)
-
-var _ PriorityAssigner = (*priorityAssignerImpl)(nil)
```

Comment:
Any reason to remove? I admit they don't do a lot, but compile-time checks are nice?

Reply:
There is no need in them, since NewPriorityAssigner returns an interface and compile time check happens there. This is only useful if you have completely separate struct, but you want to make sure that it is in sync with interface.

Reply:
yes `func NewPriorityAssigner(..) PriorityAssigner` already ensures this at compile time. 

Also there's a whole another debate to have about client side interfaces in Go. Instead of declaring interface and implementing it in the same package, the client (caller side) defines the interface it expects. https://medium.com/@mbinjamil/using-interfaces-in-go-the-right-way-99384bc69d39. We can discuss pros/cons of that another time. This pattern is not used in our repo at all.